icu4c/source/i18n/erarules.cpp

```diff
@@ -116,16 +116,17 @@ EraRules* EraRules::createInstance(const char *calType, UBool includeTentativeEr
     ures_getByKey(rb.getAlias(), "calendarData", rb.getAlias(), &status);
     ures_getByKey(rb.getAlias(), calType, rb.getAlias(), &status);
     ures_getByKey(rb.getAlias(), "eras", rb.getAlias(), &status);
-        
+
     if (U_FAILURE(status)) {
         return nullptr;
     }
 
     int32_t numEras = ures_getSize(rb.getAlias());
     int32_t firstTentativeIdx = MAX_INT32;
 
-    LocalArray<int32_t> startDates(new int32_t[numEras], status);
-    if (U_FAILURE(status)) {
+    LocalMemory<int32_t> startDates(reinterpret_cast<int32_t *>(uprv_malloc(numEras * 4)));
```

Comment:
What is the `* 4`? Is that for the size of int32_t? In that case, `* sizeof int32_t` instead would make that much more clear.

Reply:
Re the info output: These dependencies on system symbols are needed or not needed based on the compiler flags used. If I remove them, then it will cause an error for someone with different flags; so I let these accumulate. icuplug depends on platform when it's enabled, but I don't build ICU with it enabled.

Reply:
Since we use fixed-size types, I don't find sizeof(int32_t) better than "4", but I changed it.

Reply:
Oh, and in an unoptimized build you only get half as many info messages. You get more in an optimized build because more system stuff gets inlined.